storage/key_index.go

```diff
@@ -37,115 +38,125 @@ var (
 // compact(2) on the previous example
 // generations:
 //    {empty}
-//    {4, 5(t)}
-//    {2, 3(t)}
+//    {4.0, 5.0(t)}
+//    {2.0, 3.0(t)}
 //
 // compact(4)
 // generations:
 //    {empty}
-//    {4, 5(t)}
+//    {4.0, 5.0(t)}
 //
 // compact(5):
 // generations:
-//    {empty}
-//    {5(t)}
+//    {empty} -> key SHOULD be removed.
 //
 // compact(6):
 // generations:
 //    {empty} -> key SHOULD be removed.
 type keyIndex struct {
 	key         []byte
-	index       uint64
+	rev         int64
 	generations []generation
 }
 
-// put puts an index to the keyIndex.
-func (ki *keyIndex) put(index uint64) {
-	if index < ki.index {
-		log.Panicf("store.keyindex: put with unexpected smaller index [%d / %d]", index, ki.index)
+// put puts a reversion to the keyIndex.
+func (ki *keyIndex) put(rev int64, subrev int64) {
```

Comment:
switch between `atRev int64` in Get, `rev reversion` in Put and `rev, subrev int64` here makes it complicated..